rest-api/rest-api.php

```diff
@@ -0,0 +1,672 @@
+                if ( ( $value !== 'all' ) && ! in_array( $value, $post_types ) ) {
+                    return new WP_Error( 'rest_invalid_param', sprintf( '%1$s is not a valid post type', $value ), array( 'status' => 400 ) );
+                }
+            }
+        } else {
+                // This code won't execute because we have specified this argument as required.
+                // If we reused this validation callback and did not have required args then this would fire.
+                return new WP_Error( 'rest_invalid_param', sprintf( '%s was not registered as a request argument.', $param ), array( 'status' => 400 ) );
+        }
+
+        // If we got this far then the data is valid.
+        return true;
+    }
+
+    public function add_location( $request ) {
+
+        $url_params = $request->get_url_params();
+        $body = $request->get_json_params() ?? $request->get_body_params();
+
+        dt_write_log( ' request ' );
+        dt_write_log( $request );
+       // $result = DT_Posts::geolocate_addresses( $body['id'], $body['post_type'], 'country', $body['country'] );
+
+        return $result;
+    }
+
+    public function map_fields_to_contact( $contact_map ) {
+
+        $fields_map = null;
+
+        $fields_map = array(
+        'title' => $contact_map['name'],
+        'type' => 'access',
+        'milestones' => [],
+        'maarifa_data' => $contact_map['id'],
+
+        );
+
+        $fields_map['sources']['values'][0]['value'] = 'maarifa';
+
+        $fields_map['maarifa_data'] = $contact_map;
+
+
+        if ( !empty( $contact_map['email'] ) ) {
+
+            $str_arr = preg_split( '/\,/', $contact_map['email'] );
+
+            $count = count( $str_arr );
+            for ( $i = 0; $i < $count; $i++ )
+            {
+                        $fields_map['contact_email']['values'][$i]['value'] = trim( trim( trim( $str_arr[$i], '[' ), ']' ), '"' );
+
+            }
+
+        }
+
+        if ( !empty( $contact_map['phone'] ) ) {
+
+            //$fields_map['contact_phone'] = [ [ 'value' => $contact_map['phone'] ] ];
+
+            $str_arr = preg_split( '/\,/', $contact_map['phone'] );
+
+            $count = count( $str_arr );
+
+            for ( $i = 0; $i < $count; $i++ )
+            {
+                $fields_map['contact_phone']['values'][$i]['value'] = trim( trim( trim( $str_arr[$i], '[' ), ']' ), '"' );
+            }
+        }
+
+        if ( !empty( $contact_map['facebook'] ) ) {
+
+            $fields_map['contact_facebook'] = [ [ 'value' => $contact_map['facebook'] ] ];
+        }
+
+        if ( !empty( $contact_map['gender'] ) ) {
+
+            $fields_map['gender'] = $contact_map['gender'];
+        }
+
+        if ( !empty( $contact_map['street'] ) ) {
+
+            dt_write_log( 'Valor street' );
+            dt_write_log( $contact_map['street'] );
+
+            $fields_map['contact_address'] = [ [ 'value' => $contact_map['country'] ] ];
+        }
```

Comment:
This if statement uses `street` in the condition, but then uses `country` to set the address. I think you want country in the condition so that they match.